packages/jupyterlab-gridstack/src/editor/gridstack/gridstackModel.ts

```diff
@@ -268,26 +272,50 @@ export class GridStackModel {
         cell.appendChild(item.node);
         break;
       }
-      case 'markdown':
-        renderMarkdown({
-          host: cell,
-          source: cellModel.value.text,
-          sanitizer: this.rendermime.sanitizer,
-          latexTypesetter: this.rendermime.latexTypesetter,
-          linkHandler: this.rendermime.linkHandler,
-          resolver: this.rendermime.resolver,
-          shouldTypeset: false,
-          trusted: true
+      case 'markdown': {
+        const markdownCell = new MarkdownCell({
+          model: cellModel as MarkdownCellModel,
+          rendermime: this.rendermime,
+          contentFactory: this.contentFactory,
+          editorConfig: this._editorConfig.markdown,
+          updateEditorOnShow: false
         });
+        markdownCell.inputHidden = false;
+        markdownCell.rendered = true;
+        const collapser = markdownCell.node.getElementsByClassName(
+          'jp-Collapser'
+        );
+        for (let i = 0; i < collapser.length; i++) {
+          collapser[i].remove();
+        }
+        const prompt = markdownCell.node.getElementsByClassName(
+          'jp-InputPrompt'
+        );
+        for (let i = 0; i < prompt.length; i++) {
+          prompt[i].remove();
+        }
+        cell.appendChild(markdownCell.node);
         break;
-
-      default:
-        renderText({
-          host: cell,
-          source: cellModel.value.text,
-          sanitizer: this.rendermime.sanitizer
+      }
+      default: {
+        const rawCell = new RawCell({
+          model: cellModel as RawCellModel,
+          contentFactory: this.contentFactory,
+          editorConfig: this._editorConfig.raw,
+          updateEditorOnShow: false
         });
+        rawCell.inputHidden = false;
+        const collapser = rawCell.node.getElementsByClassName('jp-Collapser');
+        for (let i = 0; i < collapser.length; i++) {
+          collapser[i].remove();
+        }
+        const prompt = rawCell.node.getElementsByClassName('jp-InputPrompt');
+        for (let i = 0; i < prompt.length; i++) {
+          prompt[i].remove();
+        }
```

Comment:
Since this logic looks similar for both the raw and markdown cells, wondering whether it could be extracted into its own function in a `Private` namespace to the module?

Reply:
Sure, I'm on it!